### ISSUE TITLE
Compilation error: ‘CARLA_EXPORT’ does not name a type

### ISSUE THREAD

**Issue by djnotes** [LMMS/lmms#6395]
### Bug Summary

I am not able to build master on Fedora 35, though all dependencies have presumably been installed. 

#### Steps to reproduce
make directory `build` and cd into it. Run `cmake ..` and then `make -j 4`

#### Expected behavior
Should make without problem. 

#### Actual behavior

Shows this: 

```
[  8%] Building CXX object plugins/CrossoverEQ/CMakeFiles/crossovereq.dir/CrossoverEQControls.cpp.o
In file included from /home/mehdi/src/lmms/plugins/CarlaBase/Carla.cpp:25:
/home/mehdi/src/lmms/plugins/CarlaBase/Carla.h:45:5: error: ‘CARLA_EXPORT’ does not name a type; did you mean ‘CARLA_API_EXPORT’?
   45 |     CARLA_EXPORT
      |     ^~~~~~~~~~~~
      |     CARLA_API_EXPORT
/home/mehdi/src/lmms/plugins/CarlaBase/Carla.h:48:5: error: ‘CARLA_EXPORT’ does not name a type; did you mean ‘CARLA_API_EXPORT’?
   48 |     CARLA_EXPORT
      |     ^~~~~~~~~~~~
      |     CARLA_API_EXPORT
/home/mehdi/src/lmms/plugins/CarlaBase/Carla.cpp: In constructor ‘CarlaInstrument::CarlaInstrument(InstrumentTrack*, const Plugin::Descriptor*, bool)’:
/home/mehdi/src/lmms/plugins/CarlaBase/Carla.cpp:152:32: error: ‘carla_get_native_patchbay_plugin’ was not declared in this scope
  152 |       fDescriptor(isPatchbay ? carla_get_native_patchbay_plugin() : carla_get_native_rack_plugin()),
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mehdi/src/lmms/plugins/CarlaBase/Carla.cpp:152:69: error: ‘carla_get_native_rack_plugin’ was not declared in this scope
  152 |       fDescriptor(isPatchbay ? carla_get_native_patchbay_plugin() : carla_get_native_rack_plugin()),
      |                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [plugins/CarlaBase/CMakeFiles/carlabase.dir/build.make:91: plugins/CarlaBase/CMakeFiles/carlabase.dir/Carla.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2401: plugins/CarlaBase/CMakeFiles/carlabase.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  8%] Building CXX object plugins/CrossoverEQ/CMakeFiles/crossovereq.dir/CrossoverEQControlDialog.cpp.o
[  8%] Building CXX object plugins/CrossoverEQ/CMakeFiles/crossovereq.dir/moc_CrossoverEQControls.cpp.o
[  8%] Building CXX object plugins/CrossoverEQ/CMakeFiles/crossovereq.dir/moc_CrossoverEQControlDialog.cpp.o
[  8%] Building CXX object plugins/CrossoverEQ/CMakeFiles/crossovereq.dir/qrc_crossovereq.cpp.o
[  8%] Linking CXX shared module ../libcrossovereq.so
[  8%] Built target crossovereq
AutoMoc: /home/mehdi/src/lmms/include/MidiApple.h:0: Note: No relevant classes found. No output generated.
AutoMoc: /home/mehdi/src/lmms/include/MidiSndio.h:0: Note: No relevant classes found. No output generated.
AutoMoc: /home/mehdi/src/lmms/include/MidiWinMM.h:0: Note: No relevant classes found. No output generated.
[  8%] Built target lmmsobjs_autogen
make: *** [Makefile:156: all] Error 2

```

#### Affected LMMS versions
1.3.0-alpha.1.184+g3964c53a0

**Comment by TheKozax** [comment]
same issue on archlinux

**Comment by djnotes** [comment]
I was able to work around the error by changing `CARLA_EXPORT` to `CARLA_API_EXPORT` inside `/lmms/plugins/CarlaBase/Carla.h` (line 45 & 48) following the error's hint

**Comment by thmo** [comment]
Shouldn't this be changed in the LMMS repo as well?

Maybe with some `#ifdef` for Carla >= 2.4.3 (changed here: https://github.com/falkTX/Carla/commit/b4cf0412 if I'm not mistaken).

**Comment by PhysSong** [comment]
Fixed in #6403.
